### PR TITLE
[OJ-37307] Add Updated Black/iSort To Pre-commit

### PR DESCRIPTION
**Pull request by jruel4**
- Update black to 24.2.0, added it as a dev dependency to pyproject
- Add isort, and a profile pyproject
- Move around pyproject.toml for standardization purposes

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,59 @@
+[project]
+name = "jf_agent"
+version = "0.0.1"
+description = "An agent that can run on-premise to download and send data to Jellyfish."
+authors = [
+    {name = "jellyfish-oss", email = "oss@jellyfish.co"},
+]
+dependencies = [
```
Comment (jruel4) [inline pyproject.toml:8]: These are kept the same, except we moved black into the dev dependencies

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+    "pytest>=7.4.0",
+    "black==23.12.0",
+    "isort==5.12.0",
+    "pre-commit>=3.6.0",
+]
+
+[tool.pdm.options]
+lock = ["--strategy", "inherit_metadata", "--python", ">=3.10,<3.11"]
+
+[tool.isort]
```
Comment (jruel4) [inline pyproject.toml:50]: Added this for isort

File: pyproject.toml [hunk 3]
```diff
@@ -1,7 +1,59 @@
...
+profile = "black"
+
 [tool.black]
 line-length = 100
 skip-string-normalization = true
-target_version = ['py37', 'py38']
+target_version = ['py310']
```
Comment (jruel4) [inline pyproject.toml:56]: Updated target version

File: README.md [hunk 4]
```diff
@@ -35,3 +35,11 @@ Here's how to fix it:
         jellyfishco/jf_agent:stable \
         $@
     ```
+
+## Development
```
Comment (jruel4) [inline README.md:39]: Added dev section

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,13 +1,19 @@
 repos:
   - repo: https://github.com/ambv/black
-    rev: 19.10b0
+    rev: 23.12.0
```
Comment (jruel4) [inline .pre-commit-config.yaml:3]: Updated to match jf_ingest, also pinned pdm to this version

**Review by lheureuxe13 — APPROVED** [review]

File: pyproject.toml [hunk 6]
```diff
@@ -1,7 +1,59 @@
...
+url = "https://pypi.python.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[tool.pdm.dev-dependencies]
+dev = [
+    "requests-mock",
+    "pytest>=7.4.0",
+    "black==23.12.0",
+    "isort==5.12.0",
```
Comment (herbal-jelly) [inline pyproject.toml:43]: Is there a compatibility matrix for black + others... what issues did you run into that keeps us from using the latest?

Comment (jruel4) [reply pyproject.toml]: Updated

**Pull request merged**